src/plugins/Makefile.am

```diff
@@ -223,6 +234,23 @@ libgs_plugin_packagekit_proxy_la_LIBADD = $(GS_PLUGIN_LIBS)
 libgs_plugin_packagekit_proxy_la_LDFLAGS = -module -avoid-version
 libgs_plugin_packagekit_proxy_la_CFLAGS = $(GS_PLUGIN_CFLAGS) $(WARN_CFLAGS)
 
+if HAVE_EAM
+eos-app-manager-service.h: Makefile $(top_srcdir)/data/eam-service-interface.xml
+	$(AM_V_GEN) $(MKDIR_P) lib && \
+	$(GDBUS_CODEGEN) \
+		--interface-prefix com.endlessm. \
+		--generate-c-code eos-app-manager-service \
+		--c-generate-object-manager \
+		--c-namespace Eos \
+		$(top_srcdir)/data/eam-service-interface.xml
+eos-app-manager-service.c: eos-app-manager-service.h
+    @: # generated as a side-effect
+
+EXTRA_DIST = $(top_srcdir)/data/eam-service-interface.xml
+CLEANFILES = eos-app-manager-service.c eos-app-manager-service.h
+BUILT_SOURCES = eos-app-manager-service.c eos-app-manager-service.h
+endif
+
 check_PROGRAMS =					\
 	gs-self-test
```

Comment:
You removed `gs-self-test` from `TESTS`, but I think you want to remove it from `check_PROGRAMS` as well, since that's the target used for `make check` and `make distcheck`.